src/Umbraco.Infrastructure/Services/Implement/CacheInstructionService.cs

```diff
@@ -0,0 +1,459 @@
+            }
+
+            return numberOfInstructionsProcessed;
+        }
+
+        /// <summary>
+        /// Attempts to deserialize the instructions to a JArray.
+        /// </summary>
+        private bool TryDeserializeInstructions(CacheInstruction instruction, out JArray jsonInstructions)
+        {
+            try
+            {
+                jsonInstructions = JsonConvert.DeserializeObject<JArray>(instruction.Instructions);
+                return true;
+            }
+            catch (JsonException ex)
+            {
+                _logger.LogError(ex, "Failed to deserialize instructions ({DtoId}: '{DtoInstructions}').",
+                    instruction.Id,
+                    instruction.Instructions);
+                jsonInstructions = null;
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Parses out the individual instructions to be processed.
+        /// </summary>
+        private static List<RefreshInstruction> GetAllInstructions(IEnumerable<JToken> jsonInstructions)
+        {
+            var result = new List<RefreshInstruction>();
+            foreach (JToken jsonItem in jsonInstructions)
+            {
+                // Could be a JObject in which case we can convert to a RefreshInstruction.
+                // Otherwise it could be another JArray - in which case we'll iterate that.
+                if (jsonItem is JObject jsonObj)
+                {
+                    RefreshInstruction instruction = jsonObj.ToObject<RefreshInstruction>();
+                    result.Add(instruction);
+                }
+                else
+                {
+                    var jsonInnerArray = (JArray)jsonItem;
+                    result.AddRange(GetAllInstructions(jsonInnerArray)); // recurse
+                }
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Processes the instruction batch and checks for errors.
+        /// </summary>
+        /// <param name="processed">
+        /// Tracks which instructions have already been processed to avoid duplicates
+        /// </param>
+        /// Returns true if all instructions in the batch were processed, otherwise false if they could not be due to the app being shut down
+        /// </returns>
+        private bool ProcessDatabaseInstructions(IReadOnlyCollection<RefreshInstruction> instructionBatch, CacheInstruction instruction, HashSet<RefreshInstruction> processed, bool released, ref int lastId)
+        {
+            // Execute remote instructions & update lastId.
+            try
+            {
+                var result = NotifyRefreshers(instructionBatch, processed, released);
+                if (result)
+                {
+                    // If all instructions were processed, set the last id.
+                    lastId = instruction.Id;
+                }
+
+                return result;
+            }
+            //catch (ThreadAbortException ex)
```

Comment:
Should this be removed?

Reply:
Probably.  I didn't add it, but agree it doesn't seem to be useful.  Will remove.